### PR TITLE
Support returning failed input descriptors on cred application validation

### PR DESCRIPTION
**Pull request by decentralgabe**

### CODE REVIEW

File: credential/manifest/model.go [hunk 1]
```diff
@@ -209,130 +212,151 @@ func IsValidCredentialApplicationForManifest(cm CredentialManifest, applicationA
...
-	// The Credential Application object MUST contain a presentation_submission property IF the related Credential Manifest contains a presentation_definition.
-	// Its value MUST be a valid Presentation Submission:
-	if !cm.PresentationDefinition.IsEmpty() {
-		if ca.PresentationSubmission.IsEmpty() {
-			return errresp.NewErrorResponse(errresp.ApplicationError, "credential application's presentation submission cannot be empty because the credential manifest's presentation definition is not empty")
-		}
+	// The Credential Application object MUST contain a presentation_submission property IF the related Credential
+	// Manifest contains a presentation_definition. Its value MUST be a valid Presentation Submission:
+	if cm.PresentationDefinition.IsEmpty() {
+		return
```
Comment (nitro-neal) [inline credential/manifest/model.go:231]: good call, 

I wonder if there is an edge case where if there is not a PresDef but the presentation submission exists (for some reason?) and then it is invalid

File: error/response.go [hunk 2]
```diff
@@ -11,7 +11,7 @@ type (
 )
 
 const (
-	ApplicationError Type = "ApplicationError"
+	ApplicationError Type = "Valid"
```
Comment (nitro-neal) [inline error/response.go:14]: do you want to rename ApplicationError to ValidError?

**Review by nitro-neal — APPROVED** [review]

File: credential/manifest/model.go (hunk 1 above)
Comment (decentralgabe) [reply credential/manifest/model.go:231]: great point - will add a check

File: error/response.go (hunk 2 above)
Comment (decentralgabe) [reply error/response.go:14]: whoops, will revert

**Pull request merged**